packages/quri_parts/jijmodeling_transpiler_quantum/quri_parts/qaoa/to_qaoa_quri.py

```diff
@@ -0,0 +1,124 @@
+from jijmodeling_transpiler_quantum.core import qubo_to_ising
+from .ising_hamiltonian_quri import to_ising_operator_from_qubo_quri
+from quri_parts.circuit import LinearMappedUnboundParametricQuantumCircuit
+from quri_parts.core.operator import Operator
+from math import pi
+
+
+class QuriQAOAAnsatzBuilder:
+    def __init__(
+        self,
+        pubo_builder: jmt.core.pubo.PuboBuilder,
+        num_vars: int,
+        compiled_instance: jmt.core.CompiledInstance,
+    ):
+        self.pubo_builder = pubo_builder
+        self.num_vars = num_vars
+        self.compiled_instance = compiled_instance
+
+    @property
+    def var_map(self) -> dict[str, tuple[int, ...]]:
+        return self.compiled_instance.var_map.var_map
+
+    def get_hamiltonian(
+        self,
+        multipliers: dict = None,
+        detail_parameters: dict = None,
+    ) -> tuple[Operator, float]:
+        qubo, constant = self.pubo_builder.get_qubo_dict(
+            multipliers=multipliers, detail_parameters=detail_parameters
+        )
+        ising_operator, ising_const = to_ising_operator_from_qubo_quri(
+            qubo, self.num_vars
+        )
+        return ising_operator, ising_const + constant
+
+    def get_qaoa_ansatz(
+        self,
+        p: int,
+        multipliers: dict = None,
+        detail_parameters: dict = None,
+    ) -> tuple[UnboundParametricQuantumCircuit, operator, float]:
+        ising_operator, constant = self.get_hamiltonian(
+            multipliers=multipliers, detail_parameters=detail_parameters
+        )
+        pauli_terms = []
+        coeff_terms = []
+        pauli_z = []
+        pauli_zz = []
+        pauli_z_coeff = []
+        pauli_zz_coeff = []
+
+        for i in range(ising_operator.n_terms - 1):
+            keys = list(ising_operator.keys())[i]
+            pauli_terms.append(keys.index_and_pauli_id_list[0])
+            items = list(ising_operator.items())[i]
+            coeff_terms.append(items[1])
+
+        quri_QAOAANSATZ = LinearMappedUnboundParametricQuantumCircuit(self.num_vars)
+
+        pauli_z = pauli_terms[: self.num_vars]
+        pauli_zz = pauli_terms[self.num_vars :]
+        pauli_zz = [sorted(sublist) for sublist in pauli_zz]
+        pauli_z_coeff = coeff_terms[: self.num_vars]
+        pauli_zz_coeff = coeff_terms[self.num_vars :]
+
+        for i in pauli_z:
+            quri_QAOAANSATZ.add_H_gate(i[0])
+
+        for i in range(p):
+            Gamma = quri_QAOAANSATZ.add_parameters(f"gamma{i}")
+            Beta = quri_QAOAANSATZ.add_parameters(f"beta{i}")
+
+            for i in zip(pauli_z, pauli_z_coeff):
+                quri_QAOAANSATZ.add_ParametricRZ_gate(i[0][0], {Gamma[0]: 2 * i[1]})
+
+            for i in zip(pauli_zz, pauli_zz_coeff):
+                quri_QAOAANSATZ.add_CNOT_gate(i[0][0], i[0][1])
+                quri_QAOAANSATZ.add_ParametricRZ_gate(i[0][1], {Gamma[0]: 2 * i[1]})
+                quri_QAOAANSATZ.add_CNOT_gate(i[0][0], i[0][1])
```

Comment:
I think there exists a function in which you can create Rzz directly.

Reply:
For the Parametric_gate I didn't find the Rzz gate.

Reply:
I guess you can use this function
https://quri-parts.qunasys.com/quri_parts/circuit/quri_parts.circuit.html#quri_parts.circuit.UnboundParametricQuantumCircuit.add_ParametricPauliRotation_gate

This may be prametrized version of Pauli rotation gate
https://quri-parts.qunasys.com/quri_parts/circuit/quri_parts.circuit.gates.html#quri_parts.circuit.gates.PauliRotation

Reply:
https://github.com/Jij-Inc/JijModelingTranspilerQuantum/blob/4194d3d1901961a44854d91b809e23eca21b6370/packages/quri_parts/jijmodeling_transpiler_quantum/quri_parts/qaoa/to_qaoa.py#L73C14-L75
Fixed!